pkg/cache/v3/linear.go

```diff
@@ -131,6 +153,35 @@ func (cache *LinearCache) notifyAll(modified map[string]struct{}) {
 		cache.respond(value, nil)
 	}
 	cache.watchAll = make(watches)
+
+	cache.updateVersionMap(modified)
+
+	for id, watch := range cache.deltaWatches {
+		res := cache.respondDelta(watch.Request, watch.Response, watch.StreamState)
+		if res != nil {
+			delete(cache.deltaWatches, id)
+		}
+	}
+}
+
+func (cache *LinearCache) respondDelta(request *DeltaRequest, value chan DeltaResponse, state stream.StreamState) *RawDeltaResponse {
+	resources := &resourceContainer{
+		resourceMap:   cache.resources,
+		versionMap:    cache.versionMap,
+		systemVersion: cache.getVersion(),
+	}
```

Comment:
@alecholmez in order to get rid of the cacheWrapper, we now initialize resourceContainer struct separately for the simple and linear caches. And we still reuse createDeltaResponse function, which now take resourceContainer struct as an argument.

Reply:
That sounds good to me